### PR TITLE
[2.19.x] DDF-6254 Hide filters in search forms with no support

### PR DESCRIPTION
**Pull request by lavoywj**
#### What does this PR do?
Currently for search forms, only the `AND` & `NOT` operators have support implemented for them but `NOT AND` & `NOT OR` still show up as options in the selection dropdown, causing errors when trying to save forms that include those operators. This PR hides the operators that have no support implemented for them so they can't be selected for use on a search form.
#### Who is reviewing it? 
@bennuttle @leo-sakh @jMoneee @frnkshin 

#### How should this be tested?
1. Build / run DDF, install profile, and navigate to Search Forms
2. Create a new search form, verify the filter options dropdown only includes `AND` & `OR` options
3. Navigate to Intrigue, create a new advanced query, verify the filter options dropdown includes `AND`, `OR`, `NOT AND`, &`NOT OR` options
#### Any background context you want to provide?
Support for `NOT AND` / `NOT OR` in search forms could be implemented later on, this just prevents the use of them being used incorrectly 
#### What are the relevant tickets?
Fixes: #6254 

#### Screenshots
Before: 
Screen Shot 2020-08-19 at 6 54 53 AM

After:
Screen Shot 2020-08-19 at 6 56 15 AM

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.view.js [hunk 1]
```diff
@@ -59,29 +59,47 @@ module.exports = Marionette.LayoutView.extend({
...
+              label: 'OR',
+              value: 'OR',
+            },
+          ],
+          defaultSelection: [this.model.get('operator') || 'AND'],
+        })
+      )
+    } else {
+      this.filterOperator.show(
+        DropdownView.createSimpleDropdown({
```
Comment (leo-sakh) [inline ui/packages/catalog-ui-search/src/main/webapp/component/filter-builder/filter-builder.view.js:80]: ✏️ There's a bit of duplication here. I recommend encapsulating the common bits. One approach:
```
const extraLabels = [
              {
              label: 'NOT AND',
              value: 'NOT AND',
            },
            {
              label: 'NOT OR',
              value: 'NOT OR',
            },
]

const showExtraLabels = !this.options || this.options.isFormBuilder !== true

const labels = {
          list: [
            {
              label: 'AND',
              value: 'AND',
            },
            {
              label: 'OR',
              value: 'OR',
            },
            ...(showExtraLabels ? extraLabels : [])
        ]
}
```

**Comment by bdeining** [comment]
build now

**Comment by lavoywj** [comment]
build now

**Comment by bdeining** [comment]
hero looks good

**Pull request merged**